core/kg-query/src/main/java/it/unibz/inf/ontop/query/translation/impl/RDF4JTupleExprTranslator.java

```diff
@@ -219,38 +222,73 @@ private TranslationResult translate(Group group) throws OntopInvalidKGQueryExcep
-            ImmutableTerm term = translator.getTerm(elem.getOperator());
+            RDF4JValueExprTranslator.ExtendedTerm term = getValueTranslator(allowedVars).getTerm(elem.getOperator());
+
             Variable definedVar = termFactory.getVariable(elem.getName());
             allowedVars.add(definedVar);
 
-            result.add(new VarDef(definedVar, term));
+            result.add(new VarDef(definedVar, term.getTerm()));
+            existsBuilder.putAll(term.getExistsMap());
         }
-        return mergeVarDefs(ImmutableList.copyOf(result));
+        return new ValueExpressionResult<>(mergeVarDefs(ImmutableList.copyOf(result)), existsBuilder.build());
     }
 
     private TranslationResult translate(Order order) throws OntopInvalidKGQueryException, OntopUnsupportedKGQueryException {
         TranslationResult child = translate(order.getArg());
         RDF4JValueExprTranslator translator = getValueTranslator(child.iqTree.getVariables());
 
-        ImmutableList<OrderByNode.OrderComparator> comparators = order.getElements().stream()
-                .map(o -> Optional.of(translator.getTerm(o.getExpr()))
-                        .filter(t -> t instanceof NonGroundTerm)
-                        .map(t -> (NonGroundTerm)t)
-                        .map(t -> iqFactory.createOrderComparator(t, o.isAscending())))
+        ImmutableMap<OrderElem, Optional<RDF4JValueExprTranslator.ExtendedTerm>> orderElements = order.getElements().stream()
+                .collect(ImmutableCollectors.toMap(
+                        o -> o,
+                        o -> Optional.of(translator.getTerm(o.getExpr()))
+                ));
+
+        ImmutableList<OrderByNode.OrderComparator> comparators = orderElements.entrySet().stream()
+                .map(e -> e.getValue()
+                        .filter(t -> t.getTerm() instanceof NonGroundTerm)
+                        .map(t -> (NonGroundTerm) t.getTerm())
+                        .map(t -> iqFactory.createOrderComparator(t, e.getKey().isAscending())))
                 .flatMap(Optional::stream)
                 .collect(ImmutableCollectors.toList());
 
+        var existsMaps = orderElements.values().stream()
+                .filter(Optional::isPresent)
+                .map(Optional::get)
+                .map(RDF4JValueExprTranslator.ExtendedTerm::getExistsMap)
+                .filter(map -> !map.isEmpty())
+                .collect(ImmutableCollectors.toList());
+
+        IQTree childTree;
+        if (existsMaps.isEmpty()) {
+            childTree = child.iqTree;
+        } else {
+            ImmutableList<IQTree> orderElementsExistsSubtrees = existsMaps.stream()
+                    .map(e -> {
+                        try {
+                            return translateExists(e, child);
+                        } catch (OntopUnsupportedKGQueryException | OntopInvalidKGQueryException ex) {
+                            throw new RuntimeException(ex);
```

Comment:
Please do not wrap the exception into a RuntimeException. It is important we preserve the checked OntopUnsupportedKGQueryException. Use a for loop instead.